wire-ios-data-model/Tests/MLS/MLSServiceTests.swift

```diff
@@ -2337,6 +2337,47 @@ final class MLSServiceTests: ZMConversationTestsBase, MLSServiceDelegate {
         XCTAssertEqual(mockConversationEventProcessor.processConversationEvents_Invocations, [])
     }
 
+    func test_UpdateKeyMaterial_ContinuesOnFailureForSomeGroups() async throws {
+        // Given
+        let group1 = MLSGroupID.random()
+        let group2 = MLSGroupID.random()
+        let group3 = MLSGroupID.random()
+
+        mockStaleMLSKeyDetector.groupsWithStaleKeyingMaterial = [group1, group2, group3]
+
+        var updatedGroups = [MLSGroupID]()
+        mockMLSActionExecutor.mockUpdateKeyMaterial = { groupID in
+            if groupID == group2 {
+                throw CommitError.failedToSendCommit(recovery: .giveUp, cause: .mlsStaleMessage)
```

Comment:
suggestion: add more context to the test as explain in PR
```suggestion
                // Given one of the group fails
                throw CommitError.failedToSendCommit(recovery: .giveUp, cause: .mlsStaleMessage)
```